doc/edgehub-api.md

```diff
@@ -0,0 +1,434 @@
+
+The structure of the message and the type of system properties can be added is described on the [IoT Hub documentation](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-construct).
+Telemetry messages are subject to [Routing](https://docs.microsoft.com/en-us/azure/iot-edge/module-composition?view=iotedge-2020-11#declare-routes). When telemetry messages are routed to an [Edge Module](https://docs.microsoft.com/en-us/azure/iot-edge/module-development?view=iotedge-2020-11), we are talking about Module-to-Module messages.
+
+If Edge Hub is between IoT Hub and a device/module, Edge Hub acts as a buffer and router. It does not just relay the messages. When a device/module sends a message, Edge Hub takes it and acknowledges it even if it cannot forward the message to IoT Hub or to another module immediately. Depending on the settings, Edge Hub stores the message in a persistent store or just holds it in memory.
+
+Then Edge Hub runs the message through its routing, and depending on the routing rules it tries to forward the message to IoT Hub or another module. If it is not possible, because Edge Hub is not in connection to IoT Hub, or the module is off, then the message stays stored and will be delivered next time it is possible.
+## Twin
+[Twins](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins) are basically a combination of configuration and status information. To understand the term "Twin" and the role of this construct, let's say that we want to control a beverage cooler and we want its temperature to set 5 Celsius.
+
+In this case the part of the twin that describes the settings/configuration we want, can look the following:
+```
+"properties": {
+    "desired": {
+        "temperature": 5
+    }
+}
+```
+After the setting has been made, it might be useful to know how the device - in this case the beverage cooler - progresses achieving the desired state. To support this scenario, twins have another section where devices can report their status:
+```
+"properties": {
+    "desired": {
+        "temperature": 5
+    },
+    "reported": {
+        "temperature": 7
+    }
+}
+```
+The device can periodically report its status and many times the desired section and the reported section converges and finally take the same values. Hence the name Twin.
+
+The "desired" section is usually set by a backend application, or the operator can use Azure Portal to set the twin. The "reported" part is usually generated by the device/module.
+
+Note, that it is not necessary that the "desired" and "reported" sections have the same properties. It is possible that the "desired" section contains a property which later is not going to be reported back. Also, it can happen that a device reports status information that does not have a corresponging part in the "desired" section.
```

Comment:
```suggestion
Note, that it is not necessary that the "desired" and "reported" sections have the same properties. It is possible that the "desired" section contains a property which later is not going to be reported back. Also, it can happen that a device reports status information that does not have a corresponding part in the "desired" section.
```